src/runtime/pkg/katautils/create.go

```diff
@@ -247,10 +263,22 @@ func CreateContainer(ctx context.Context, sandbox vc.VCSandbox, ociSpec specs.Sp
 		return vc.Process{}, err
 	}
 
-	// Run pre-start OCI hooks.
-	err = EnterNetNS(sandbox.GetNetNs(), func() error {
+	hooksFunc := func() error {
+		if err := CreateRuntimeHooks(ctx, ociSpec, containerID, bundlePath); err != nil {
+			return err
+		}
 		return PreStartHooks(ctx, ociSpec, containerID, bundlePath)
-	})
+	}
+
+	netNs := sandbox.GetNetworkNamespace()
+	// Run pre-start/createRuntime OCI hooks.
+	if netNs.NetNsCreated {
+		// network namespace created by kata, hooks run in host namespace.
+		err = hooksFunc()
+	} else {
+		err = EnterNetNS(netNs.NetNsPath, hooksFunc)
+	}
```

Comment:
But that is not correct. Both `createRuntime` and the deprecated `preStart` should always run in the runtime netns. Never in the sandbox one.